### PR TITLE
fix: frontend hotdeploy always false in production mode

### PR DESCRIPTION
**Pull request by caalador**

### CODE REVIEW

File: flow-server/src/main/java/com/vaadin/flow/server/DefaultDeploymentConfiguration.java [hunk 1]
```diff
@@ -259,6 +259,9 @@ public String getPushServletMapping() {
 
     @Override
     public boolean frontendHotdeploy() {
+        if (isProductionMode()) {
```
Comment (mshabarov) [inline flow-server/src/main/java/com/vaadin/flow/server/DefaultDeploymentConfiguration.java:262]: Should this check be in the `checkFrontendHotdeploy`, so the resolution is made there and here we just return the value.

**Comment by Artur-** [comment]
> [qtp1497503548-143] INFO com.vaadin.flow.server.DefaultDeploymentConfiguration - Vaadin is running in production mode.
Warning:  /themes/reusable-theme/fortawesome/icons/snowflake.svg
java.lang.IllegalStateException: Asset '/fortawesome/icons/snowflake.svg' is not found in project frontend directory, default development bundle or in the application bundle 'src/main/dev-bundle/assets/'. 
Verify that the asset is available in 'frontend/themes/reusable-theme/' directory and is added into the 'assets' block of the 'theme.json' file. 
Else verify that the dependency 'com.vaadin:vaadin-dev-bundle' is added to your project.

This should not happen either... Why is that checked in production mode

**Comment by mshabarov** [comment]
> ## Test Results
>    946 files  +   204     946 suites  +204   51m 6s [⏱️](https://github.com/EnricoMi/publish-unit-test-result-action/blob/v1.20/README.md#the-symbols) - 2m 42s 5 914 tests +2 041  5 862 [✔️](https://github.com/EnricoMi/publish-unit-test-result-action/blob/v1.20/README.md#the-symbols) +2 024  35 [💤](https://github.com/EnricoMi/publish-unit-test-result-action/blob/v1.20/README.md#the-symbols) +1  16 [❌](https://github.com/EnricoMi/publish-unit-test-result-action/blob/v1.20/README.md#the-symbols) +16  1 [🔥](https://github.com/EnricoMi/publish-unit-test-result-action/blob/v1.20/README.md#the-symbols) ±0  6 157 runs  +2 066  6 098 [✔️](https://github.com/EnricoMi/publish-unit-test-result-action/blob/v1.20/README.md#the-symbols) +2 049  42 [💤](https://github.com/EnricoMi/publish-unit-test-result-action/blob/v1.20/README.md#the-symbols) +1  16 [❌](https://github.com/EnricoMi/publish-unit-test-result-action/blob/v1.20/README.md#the-symbols) +16  1 [🔥](https://github.com/EnricoMi/publish-unit-test-result-action/blob/v1.20/README.md#the-symbols) ±0 
> 
> For more details on these failures and errors, see [this check](https://github.com/vaadin/flow/runs/11246580321).
> 
> Results for commit [9df268b](https://github.com/vaadin/flow/commit/9df268b47a9ab953b10252be21252518223dfbb7). ± Comparison against base commit [b28e8f5](https://github.com/vaadin/flow/commit/b28e8f5a5da80ea1ae408a04c6ecaa2eb01482a7).

Excellent catch! Here https://github.com/vaadin/flow/blob/b28e8f5a5da80ea1ae408a04c6ecaa2eb01482a7/flow-server/src/main/java/com/vaadin/flow/server/StaticFileServer.java#L267 we miss the check against production mode. I can make a separate PR for it.

File: flow-server/src/main/java/com/vaadin/flow/server/DefaultDeploymentConfiguration.java (hunk 1 above)
Comment (caalador) [reply flow-server/src/main/java/com/vaadin/flow/server/DefaultDeploymentConfiguration.java]: done

**Review by mshabarov — DISMISSED** [review]

**Comment by mshabarov** [comment]
https://github.com/vaadin/flow/pull/15884 
Need a tests for it.

**Review by mshabarov — APPROVED** [review]

**Pull request merged**